### ISSUE TITLE
Trace.WriteLine in AssemblyCleanup is not printed in MSTest v3.0.0

### ISSUE THREAD

**Issue by vaLP2eh** [microsoft/testfx#1448]
## Describe the bug

Methods having the **AssemblyCleanupAttribute** are not called.

Example:
```
[TestClass]
public sealed class DummyUnitTest
{
    [AssemblyInitialize]
    public static void AssemblyInit(TestContext testContext)
    {
        Trace.WriteLine(nameof(AssemblyInit));
    }

    [AssemblyCleanup]
    public static void AssemblblyCleanup()
    {
        Trace.WriteLine(nameof(AssemblblyCleanup));
    }

    [TestMethod]
    public void DummyTest()
    {
        Trace.WriteLine(nameof(this.DummyTest));
    }
}
```

Expected output (by MSTest 2.2.10):
```
...
AssemblyInit
DummyTest
AssemblblyCleanup
...
```

Actual output (by MSTest 3.0.0)
```
...
AssemblyInit
DummyTest
...
```

## Steps To Reproduce

Run the unit test as described above with MSTest v3.0.0.

## Expected behavior

The AssemblblyCleanup method should be called.

## Actual behavior

The AssemblblyCleanup method is not called.

[AB#1699540](https://devdiv.visualstudio.com/0bdbc590-a062-4c3f-b0f6-9383f67865ee/_workitems/edit/1699540)

**Comment by Evangelink** [comment]
@engyebrahim Looks like this might be linked to the change we did. Could you please improve our integration tests for this?

**Comment by Evangelink** [comment]
@vaLP2eh Thanks for raising this issue. I managed to confirm the behaviour in VS and cli. We will look into it as soon as possible.

**Comment by Evangelink** [comment]
@vaLP2eh We have investigated the issue and the assembly cleanup is correctly called, the problem is the trace log isn't correctly attached to the test. It will be fixed with v3.0.1

**Comment by Evangelink** [comment]
Fixed by #1448
